server/internal/infrastructure/mongo/request.go

```diff
@@ -65,6 +65,17 @@ func (r *Request) FindByIDs(ctx context.Context, ids id.RequestIDList) (request.
 	return filterRequests(ids, res), nil
 }
 
+func (r *Request) FindByItems(ctx context.Context, list id.ItemIDList) (request.List, error) {
+
+	filter := bson.M{
+		"items.item": bson.M{
+			"$in": list.Strings(),
+		},
+	}
+
```

Comment:
```suggestion
	filter := bson.M{
		"items.item": bson.M{
			"$in": list.Strings(),
		},
	}
```

Also add an index for `items.item`